File: C/01_Recap/03_Reduction/allreduce.c
#include <stdio.h>
#include <mpi.h>

#define TAG 100

int main(int argc, char ** argv)
{

  int rank, nproc, recv;

  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &nproc);
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);

  MPI_Allreduce(&rank);

  printf("On rank %3d MPI_Allreduce gives maximum rank as %3d\n", rank, recv);

  MPI_Finalize();

}

File: Fortran/01_Recap/03_Reduction/allreduce.f90
PROGRAM reduce

  USE mpi
  IMPLICIT NONE

  INTEGER, PARAMETER :: tag = 100

  INTEGER :: rank, recv
  INTEGER :: nproc, ierr

  CALL MPI_Init(ierr)

  CALL MPI_Comm_size(MPI_COMM_WORLD, nproc, ierr)
  CALL MPI_Comm_rank(MPI_COMM_WORLD, rank, ierr)

  CALL MPI_Allreduce(rank, , , , , , )
  PRINT *, 'On rank ', rank, ' MPI_Allreduce gives maximum rank as ', recv

  CALL MPI_Finalize(ierr)

END PROGRAM reduce
